lib/internal/Magento/Framework/HTTP/Client/Curl.php

```diff
@@ -246,6 +244,89 @@ public function post($uri, $params)
         $this->makeRequest("POST", $uri, $params);
     }
 
+    /**
+     * Make PUT request
+     *
+     * @param string $uri
+     * @param array|string $params
+     * @return void
+     */
+    public function put($uri, $params)
+    {
+        $this->makeRequest("PUT", $uri, $params);
+    }
+
+    /**
+     * Make DELETE request
+     *
+     * @param string $uri
+     * @param array|string $params
+     * @return void
+     */
+    public function delete(string $uri, array|string $params = [])
+    {
+        $this->makeRequest("DELETE", $uri, $params);
+    }
+
+    /**
+     * Make PATCH request
+     *
+     * @param string $uri
+     * @param array|string $params
+     * @return void
+     */
+    public function patch($uri, $params)
+    {
+        $this->makeRequest("PATCH", $uri, $params);
+    }
+
+    /**
+     * Make OPTIONS request
+     *
+     * @param string $uri
+     * @param array|string $params
+     * @return void
+     */
+    public function options(string $uri, array|string $params = [])
+    {
+        $this->makeRequest("OPTIONS", $uri, $params);
+    }
+
+    /**
+     * Make HEAD request
+     *
+     * @param string $uri
+     * @param array|string $params
+     * @return void
+     */
+    public function head(string $uri, array|string $params = [])
```

Comment:
I did not introduce a payload here because according to RFC9110

> The HEAD method is identical to GET except that the server MUST NOT send content in the response. 

https://www.rfc-editor.org/rfc/rfc9110.html#section-9.3.2

---

You're absolutely correct that the RFC you linked doesn't explicitly forbid a payload for HEAD requests. However, it strongly discourages it and hints at the reasons why it's generally not used:

## Semantic Ambiguity:

- HEAD is for metadata: The core purpose of HEAD is to fetch metadata about a resource without retrieving the actual data. A payload in this context has no clearly defined meaning. What should the server do with the provided data? Unlike POST (create/update) or PUT (replace), HEAD doesn't have an inherent action associated with a payload.

- Potential for Confusion: Including a payload with HEAD could lead to confusion and unpredictable behavior, as servers might interpret it in different ways. This ambiguity undermines the standardization that HTTP strives for.

## Why GET can have a payload (though discouraged):

While GET requests can technically have a payload, it's generally discouraged for similar reasons to HEAD.  The RFC states that a payload in a GET request "has no semantic meaning to the request." This means the server is free to ignore it.